theme/static/js/hs_resource_table.js

```diff
@@ -0,0 +1,725 @@
+
+    return inputString;
+}
+
+// Looks at the query strings in the searchbox and sets the values in the dropdown options
+function applyQueryStrings() {
+    var inputString = $("#resource-search-input").val();
+    // Matches occurrences of query strings. i.e.: author:mauriel
+    var regExp = /\[(type|TYPE|author|AUTHOR|subject|SUBJECT):[^\]|^\[]+]/g;
+    var occurrences = inputString.match(regExp);
+
+    var inputType = "";
+    var inputSubject = "";
+    var inputAuthor = "";
+
+    var collection = [];
+    if (occurrences) {
+        for (var item in occurrences) {
+            var content = occurrences[item].replace("[", "").replace("]", "").split(":");
+            collection.push(content);
+        }
+    }
+
+    for (var item in collection) {
+        if (collection[item][0].toUpperCase() == "TYPE") {
+            inputType = collection[item][1];
+        }
+        else if (collection[item][0].toUpperCase() == "AUTHOR") {
+            inputAuthor = collection[item][1];
+        }
+        else if (collection[item][0].toUpperCase() == "SUBJECT") {
+            inputSubject = collection[item][1];
+        }
+    }
+
+    $("#input-author").val(inputAuthor);
+    $("#input-subject").val(inputSubject);
+    $("#input-resource-type").val(inputType.toLowerCase());
+
+}
+
+// Looks at the values in the dropdown options and appends the corresponding query string to the search box.
+function typeQueryStrings () {
+    var searchInput = $("#resource-search-input");
+
+    var type = $("#input-resource-type").val();
+    var author = $("#input-author").val();
+    var subject = $("#input-subject").val();
+
+    var searchQuery = "";
+
+    if (type) {
+        searchQuery = searchQuery + " [type:" + type + "]";
+    }
+
+    if (author) {
+        searchQuery = searchQuery + " [author:" + author + "]";
+    }
+
+    if (subject){
+        searchQuery = searchQuery + " [subject:" + subject + "]";
+    }
+
+    searchQuery = searchQuery + removeQueryOccurrences(searchInput.val());
+    searchQuery = searchQuery.trim();
+    searchInput.val(searchQuery);
+}
+
+
+/*==================================================
+    Table columns
+    0 - actions
+    1 - Resource Type
+    2 - Title
+    3 - Owner
+    4 - Last Modified
+    5 - Subject
+    6 - Authors
+    7 - Permission Level
+    8 - Labels
+    9 - Favorite
+    10 - Last modified (sortable format)
+    11 - Sharing Status
+    12 - Access Grantor
+==================================================*/
+
+/* Custom filtering function which will search data for the values in the custom filter dropdown or query strings */
+$.fn.dataTable.ext.search.push (
+        function (settings, data, dataIndex) {
+            var inputString = $("#resource-search-input").val();
+            // Matches occurrences of query strings. i.e.: author:mauriel
+            var regExp = /\[(type|author|subject|TYPE|AUTHOR|SUBJECT):[^\]|^\[]+]/g;
+            var occurrences = inputString.match(regExp);
```

Comment:
@mauriel133 Would it be better to convert inputString to uppercase or lowercase and then regExp can only check the appropriate case accordingly? What if the upper and lower cases are mixed up in inputString which may not be captured in regExp currently, right?